compiletest: Remove the one thing that was checking a directive's `original_line`

Zalathar opened the pull request:
This special handling of `ignore-tidy*` was introduced during the migration to `//@` directives (#120881), and has become unnecessary after the subsequent removal of the legacy directive check (#131392).

Zalathar committed:
Remove the one thing that was checking a directive's `original_line`

Zalathar committed:
Rename `HeaderLine` to `DirectiveLine`

Zalathar commented:
Oh, looks like it was introduced *because* of the `//@` migration: #120881. So yeah, definitely not needed after removing the legacy check.

Zalathar commented:
r? jieyouxu

jieyouxu reviewed (approved):
Oh yeah good catch. Thanks!

jieyouxu commented:
You can r=me after PR CI is green.

Zalathar commented:
🟩

@bors r=jieyouxu rollup

bors commented:
:pushpin: Commit 3d15d2d520ee134ffe826093b1cf4e3d722338da has been approved by `jieyouxu`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=jieyouxu 3d15d2d520ee134ffe826093b1cf4e3d722338da -->
<!-- homu: {"type":"Approved","sha":"3d15d2d520ee134ffe826093b1cf4e3d722338da","approver":"jieyouxu","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

tgross35 commented:
Looks like this failed the rollup

@bors r-

jieyouxu commented:
Ah of course, `./x doc src/tools/compiletest` is run very late into full CI pipeline...

tgross35 commented:
That does seem like the sort of thing that could be checked somewhere in PR CI

jieyouxu commented:
> That does seem like the sort of thing that could be checked somewhere in PR CI

https://github.com/rust-lang/rust/issues/131613

Zalathar commented:
Rebased and fixed trivial doc issue [(diff)](https://github.com/rust-lang/rust/compare/b41a51ffc4087ceabf4509649a23a92cf090961f..f51088021cbb7c683c4de83c42dac2b9e4dbead2).

I'm running a side job to double-check that there are no more problems, after which I'll reapply the r+.

Zalathar commented:
@bors r=jieyouxu

bors commented:
:pushpin: Commit f51088021cbb7c683c4de83c42dac2b9e4dbead2 has been approved by `jieyouxu`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=jieyouxu f51088021cbb7c683c4de83c42dac2b9e4dbead2 -->
<!-- homu: {"type":"Approved","sha":"f51088021cbb7c683c4de83c42dac2b9e4dbead2","approver":"jieyouxu","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
